aws/src/main/java/org/apache/iceberg/aws/RESTSigV4Signer.java

```diff
@@ -1,157 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-package org.apache.iceberg.aws;
-
-import java.io.IOException;
-import java.io.UncheckedIOException;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-import org.apache.hc.core5.http.EntityDetails;
-import org.apache.hc.core5.http.Header;
-import org.apache.hc.core5.http.HttpHeaders;
-import org.apache.hc.core5.http.HttpRequest;
-import org.apache.hc.core5.http.HttpRequestInterceptor;
-import org.apache.hc.core5.http.io.entity.StringEntity;
-import org.apache.hc.core5.http.protocol.HttpContext;
-import org.apache.iceberg.exceptions.RESTException;
-import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
-import software.amazon.awssdk.auth.signer.Aws4Signer;
-import software.amazon.awssdk.auth.signer.internal.SignerConstant;
-import software.amazon.awssdk.auth.signer.params.Aws4SignerParams;
-import software.amazon.awssdk.auth.signer.params.SignerChecksumParams;
-import software.amazon.awssdk.core.checksums.Algorithm;
-import software.amazon.awssdk.http.SdkHttpFullRequest;
-import software.amazon.awssdk.http.SdkHttpMethod;
-import software.amazon.awssdk.regions.Region;
-
-/**
- * Provides a request interceptor for use with the HTTPClient that calculates the required signature
- * for the SigV4 protocol and adds the necessary headers for all requests created by the client.
- *
- * <p>See <a
- * href="https://docs.aws.amazon.com/general/latest/gr/signing-aws-api-requests.html">Signing AWS
- * API requests</a> for details about the protocol.
- */
-public class RESTSigV4Signer implements HttpRequestInterceptor {
```

Comment:
Should follow a minor version deprecation? 
https://iceberg.apache.org/contribute/?h=contri#minor-version-deprecations-discretionary

Reply:
Hi @ajantha-bhat I think the deprecation cycle does not apply here because this component is not meant to be instantiated by users directly. It was only instantiated by reflection when SigV4 was enabled:

https://github.com/apache/iceberg/blob/0bc6dfa1bd55c8abb4891f1eeadb6fbc89878c92/core/src/main/java/org/apache/iceberg/rest/HTTPClient.java#L558-L560

But I'm happy to keep this class around if others think that's better.

Reply:
ACK.